### ISSUE TITLE
Fragments are not diffed by React on route changes, if they didn't match any routes on initial page load.

### ISSUE THREAD

**Issue by MatthewPikar** [FormidableLabs/redux-little-router#179]
See issue #171 

Given fragments like so:
```
<Fragment forRoutes={ ['/foos', '/foos/:foo'] } >
  <Foos />
</Fragment>

<Fragment forRoute={ '/foos/:foo/bars/:bar' } >
  <Bar />
</Fragment>
```
If initial route leads to '/foos', the second Fragment for route '/foos/:foo/bars/:bar' won't load on a route change unless the parent component explicitly uses router's state.  React never diffs the second Fragment on subsequent route changes.

Although the workaround is as trivial always making sure the parent component loads in router's state, this doesn't seem very intuitive and might stump novice users.

**Comment by mherodev** [comment]
Hrm, I wonder if I'm running into something related to this. I am not explicitly using the router's state anywhere, and my Fragments are rendering incorrectly. When I navigate to the homepage, I see everything as I'd expect. But navigated to another page isn't unloading the homepage Fragment, so other pages are stacked below home.

I'm also not quite sure I understand why I can access `router` from the props in the root component but my Fragments don't seem to be passing `router` down to their children.

Something is amiss.

**Comment by tptee** [comment]
Hey @MatthewPikar , `Fragment` now `connect()`s directly to the router state in `master`. Can you clone and see if this fixes your issue?

PS `forRoutes` never made it over from the `AbsoluteFragment` days, so it's probably not doing anything in your example.

**Comment by tptee** [comment]
This should be fixed in [v14.0.0-0](https://github.com/FormidableLabs/redux-little-router/releases/tag/v14.0.0-0)
